src/main/java/digital/vix/vertxdemo/repository/EndpointRepositoryImpl.java

```diff
@@ -25,6 +25,22 @@ public EndpointRepositoryImpl(SQLClient sqlClient) {
 				.flatMapPublisher(resultSet -> Flowable.fromIterable(resultSet.getRows()));
 	}
 
+	@Override
+	public Flowable<JsonObject> findByEndPointIds(long[] ids) {
+		JsonArray idsPara = new JsonArray();
+		String queryExtension ="";
+		for (long id:  ids){
+			idsPara.add(id);
+			queryExtension += "?, ";
+		}
+		queryExtension = (queryExtension.length()>0) ? queryExtension.substring(0,queryExtension.length()-2) : queryExtension;
+
+		return sqlClient
+				.rxQueryWithParams("SELECT * FROM endpoints where id IN (" + queryExtension+ ")= ?", idsPara)
```

Comment:
Might aid readability if you started the SQL string at the top, and sequentially built then feed the full item into the sqlClient  

I'm a bit confused why the final string looking like:
"SELECT * FROM endpoints where id IN (?, ?, ?, ?)= ?"

I think you don't need the last few characters?
```suggestion
				.rxQueryWithParams("SELECT * FROM endpoints where id IN (" + queryExtension+ ")", idsPara)
```